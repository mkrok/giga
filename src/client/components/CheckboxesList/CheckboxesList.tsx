import { useRef } from 'react';
import styled from 'styled-components';
import Checkbox from '../Checkbox/Checkbox';
import { SIZE } from '~/config/constants';

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-evenly;
  background-color: #222;
  color: #ccc;
  padding: 1rem;
  width: 100%;
`;

const CheckboxesList = () => {
  const ref = useRef(null);
  return (
    <Wrapper>
      {Array.from(Array(SIZE).keys()).map(index => (
        <Checkbox index={index} key={`checkbox-${index + 1}`} />
      ))}
    </Wrapper>
  );
};

export default CheckboxesList;
