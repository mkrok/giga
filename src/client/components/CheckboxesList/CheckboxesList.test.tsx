import renderer from 'react-test-renderer';
import CheckboxesList from '.';

describe('CheckboxesList', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<CheckboxesList />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
