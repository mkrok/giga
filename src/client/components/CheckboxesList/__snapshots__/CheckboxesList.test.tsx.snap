// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxesList should match snapshot 1`] = `
<div
  className="sc-gueYoa gXdkCB"
>
  <div
    className="sc-beqWaB dWaeJQ"
  >
    <label>
      0
    </label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <div
    className="sc-beqWaB dWaeJQ"
  >
    <label>
      1
    </label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <div
    className="sc-beqWaB dWaeJQ"
  >
    <label>
      2
    </label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <div
    className="sc-beqWaB dWaeJQ"
  >
    <label>
      3
    </label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <div
    className="sc-beqWaB dWaeJQ"
  >
    <label>
      4
    </label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <div
    className="sc-beqWaB dWaeJQ"
  >
    <label>
      5
    </label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <div
    className="sc-beqWaB dWaeJQ"
  >
    <label>
      6
    </label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
  </div>
</div>
`;
